src/Raven.Server/Documents/Patch/IJavaScriptEngine.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Jint;
+using Jint.Native;
+using Jint.Native.Object;
+using Jint.Runtime;
+using Jint.Runtime.Interop;
+using Raven.Client.ServerWide.JavaScript;
+using V8.Net;
+using Raven.Server.Extensions.V8;
+using Raven.Server.Extensions.Jint;
+using Sparrow.Json;
+using Sparrow.Server.Utils;
+
+namespace Raven.Server.Documents.Patch
+{
+    //delegate JsValue JintFunction(JsValue self, JsValue[] args); // by some reason it doesn't fit Func<JsValue self, JsValue[] args, JsValue>
+
+    public interface IJavaScriptEngine : IDisposable
+    {
+        JavaScriptEngineType EngineType { get;  }
+        
+        JsHandle JsonStringify { get;  }
+        
+        void ForceGarbageCollection();
+
+        void TryCompileScript(string script);
+
+        void Execute(string source, string sourceName = "anonymousCode.js", bool throwExceptionOnError = true);
+
+        void ExecuteWithReset(string source, string sourceName = "anonymousCode.js", bool throwExceptionOnError = true);
+
+        void ResetCallStack();
+
+        void ResetConstraints();
+
+        JsHandle GetGlobalProperty(string propertyName);
+
+        void SetGlobalProperty(string name, JsHandle value);
+        
+        JsHandle CreateCLRCallBack(string propertyName, (Func<JsValue, JsValue[], JsValue> Jint, JSFunction V8) funcTuple, bool keepAlive = true);
```

Comment:
Jint is flexible that way that it accepts PRs I believe. I would really encourage making the main Jint project support things you need instead of maintaining a fork, but that's just me.